src/main/java/org/spdx/library/model/RelatedElementCollection.java

```diff
@@ -199,7 +201,24 @@ public boolean remove(Object o) {
 						if (relatedElement.isPresent() && 
 						        relatedElement.get().equals(o) &&
 								relationship.getRelationshipType().equals(relationshipTypeFilter)) {
-							return relationshipCollection.remove(relationship);
+							IModelStore modelStore = relationship.getModelStore();
+							String documentUri = relationship.getDocumentUri();
+							final IModelStoreLock lock = modelStore.enterCriticalSection(documentUri, false);
+							try {
+								if (relationshipCollection.remove(relationship)) {
+									try {
+										modelStore.delete(documentUri, relationship.getId());
```

Comment:
I think we are missing `return true;` below this line. Now the method will return `false` if the relationship is removed from `relationshipCollection`

Reply:
Hm... if a `SpdxInUseException` is caught, it should probably return true as well, right? Since everything is ok and the relationship was removed from `relationshipCollection`

Reply:
@nicoweidner Thanks for the catch - resolved with patest commit